packages/cozy-client/src/models/konnectorFolder.js

```diff
@@ -0,0 +1,213 @@
+ * @param  {Object} folders   Object containing a mapping from folder
+ * identifiers (ex: $administrative) to their localized values (ex:
+ * Administratif).
+ * @returns {String}           The result path
+ */
+
+export const buildFolderPath = (konnector, account, folders = {}) => {
+  const fullPath = get(
+    konnector,
+    // For now konnectors are only defining one folder in their folders array
+    'folders[0].defaultDir',
+    '$administrative/$konnector/$account'
+  )
+  // Trim `/` and avoid multiple `/` characters with regexp
+  let sanitizedPath = trim(fullPath.replace(/(\/+)/g, '/'), '/')
+  // If the konnector doesn't have any of our base dir, we set it to $administrative
+  if (!hasBaseDir(sanitizedPath)) {
+    sanitizedPath = '$administrative/' + sanitizedPath
+  }
+  /**
+   * Now that we have our sanitizedPath, we can split it in two strings
+   * * `baseDir` containing the baseDir path
+   * * `buildedSubDir` containing the rest of the path (ie the path without baseDir)
+   */
+  const baseDir = sanitizedPath.split('/', 1)
+  const buildedSubDir = buildSubDir(sanitizedPath, baseDir[0])
+
+  const renderedBaseDir = renderBaseDir(baseDir[0], folders)
+  const renderedPath = renderSubDir(buildedSubDir, {
+    // When adding a new allowed variable here, please keep documentation
+    // of `renderSubDir` function up to date.
+    konnector: konnector.name,
+    account: accountModel
+      .getAccountName(account)
+      .replace(sanitizeAccountIdentifierRx, '-')
+  })
+  return `/${renderedBaseDir}/${renderedPath}`
+}
+
+/**
+ * Check if the provided Path start withs our allowedBaseDirPath to see
+ *
+ * @param {String} path - path to test
+ * @returns {Boolean}
+ */
+const hasBaseDir = path => {
+  return allowedBaseDirVariables.some(baseDirVar => {
+    return path.startsWith(baseDirVar)
+  })
+}
+/**
+ * Base directories are directory where konnector may copy their data.
+ * They are expressed as variables which then need to be localized.
+ * Default is `$administrative`.
+ */
+const allowedBaseDirVariables = ['$administrative', '$photos']
+
+/**
+ * This method creates the subDir. We can't have an empty subDir, so we set
+ * it to our default '$konnector/$account'
+ *
+ * @param {String} fullPath String containing potentially the defaultDir
+ * @param {String} defaultDir String to remove from the fullPath
+ */
+const buildSubDir = (fullPath, defaultDir) => {
+  let buildedSubDir = fullPath.substring(defaultDir.length)
+  if (buildedSubDir === '') {
+    buildedSubDir = '$konnector/$account'
+  }
+  return buildedSubDir
+}
+
+/**
+ * Render base directory, based on given folders object.
+ * For example, it will render `$administrative` with the given value passed in
+ * folders object. We expect to find in folders a localized value.
+ *
+ * @param  {String} baseDir base directory variable, expects `$administrative`
+ * or `$photos`
+ * @param  {Object} folders Object indexing base directory variable with
```

Comment:
is the same folders object as https://github.com/cozy/cozy-client/pull/1392/commits/75e367fcd2d67967eacdf97356cbb80a47c2dcd6#r1340908505 ?

Reply:
Yes